dbt/adapters/redshift/impl.py

```diff
@@ -115,6 +115,21 @@ def valid_incremental_strategies(self):
     def timestamp_add_sql(self, add_to: str, number: int = 1, interval: str = "hour") -> str:
         return f"{add_to} + interval '{number} {interval}'"
 
+    def _get_cursor(self):
+        return self.connections.get_thread_connection().handle.cursor()
+
+    def list_schemas(self, database: str, schema: Optional[str] = None) -> List[str]:
```

Comment:
I would update the associated macro to call this, otherwise any other macro that is using the associated macro would still use the postgres macro, hence pg_ tables.